library/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts

```diff
@@ -0,0 +1,521 @@
+import { CalendarDay } from '../../models/calendar-day';
+import { CalendarService } from '../../calendar.service';
+import { FdRangeDate } from '../../models/fd-range-date';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+/** Component representing the day view of the calendar. */
+@Component({
+    selector: 'fd-calendar-day-view',
+    templateUrl: './calendar-day-view.component.html',
+    styleUrls: ['./calendar-day-view.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class CalendarDayViewComponent implements OnInit, AfterViewChecked, OnChanges {
+
+    /** @hidden */
+    shortWeekDays: string[];
+
+    /** Currently displayed month and year for days */
+    @Input()
+    public currentlyDisplayed: CalendarCurrent;
+
+    /** @hidden */
+    @HostBinding('class.fd-calendar__dates')
+    public fdCalendarDateViewClass: boolean = true;
+
+    /** Actual day grid with previous/current/next month days */
+    public dayViewGrid: CalendarDay[][];
+
+    /** The currently selected FdDate model in single mode. */
+    @Input()
+    public selectedDate: FdDate;
+
+    /** The currently selected FdDates model start and end in range mode. */
+    @Input()
+    public selectedRangeDate: FdRangeDate;
+
+    /** The day of the week the calendar should start on. 1 represents Sunday, 2 is Monday, 3 is Tuesday, and so on. */
+    @Input()
+    public startingDayOfWeek: DaysOfWeek;
+
+    /** The type of calendar, 'single' for single date selection or 'range' for a range of dates. */
+    @Input()
+    calType: CalendarType = 'single';
+
+    /** Id of the calendar. If none is provided, one will be generated. */
+    @Input() id: string;
+
+    /** Function that allows to specify which function would be called, when focus wants to escape */
+    @Input()
+    focusEscapeFunction: Function;
+
+    /** Event emitted always, when model is changed in range mode */
+    @Output()
+    public readonly selectedRangeDateChange = new EventEmitter<FdRangeDate>();
+
+    /** Event emitted always, when next month is selected, by focus */
+    @Output()
+    public readonly nextMonthSelect = new EventEmitter();
+
+    /** Event emitted always, when previous month is selected, by focus */
+    @Output()
+    public readonly previousMonthSelect = new EventEmitter();
+
+    /** Event emitted always, when model is changed in single mode */
+    @Output()
+    public readonly selectedDateChange = new EventEmitter<FdDate>();
+
+    /** @hidden */
+    newFocusedDayId: string = '';
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly onDestroy$: Subject<void> = new Subject<void>();
+
+    /**
+     * Function used to disable certain dates in the calendar.
+     * @param d Date
+     */
+    @Input()
+    disableFunction = function(d: FdDate): boolean {
+        return false;
+    };
```

Comment:
need a line between the end of the function and the start of the next function's documentation.